src/executables/util_cs.c

```diff
@@ -3296,17 +3328,41 @@ applyinfo (UTIL_FUNCTION_ARG * arg)
 	  error = la_log_page_check (local_database_name, log_path, pageid, check_applied_info, check_copied_info,
 				     check_replica_info, verbose, &copied_eof_lsa, &copied_append_lsa,
 				     &applied_final_lsa);
+	  if (error != NO_ERROR)
+	    {
+	      goto print_applyinfo_usage;
+	    }
 	}
 
     check_applied_info_end:
       if (error != NO_ERROR)
 	{
 	  PRINT_AND_LOG_ERR_MSG ("%s\n", db_error_string (3));
+	  goto print_applyinfo_usage;
 	}
       error = NO_ERROR;
 
       if (check_master_info)
 	{
+	  check_master_info = false;
+	  for (; ha_node_list_pp[ha_node_cnt] != NULL;)
+	    {
+	      if (!strcmp (master_node_name, ha_node_list_pp[ha_node_cnt]))
+		{
+		  check_master_info = true;
+		  break;
+		}
+	      ha_node_cnt++;
+	    }
+	  if (!check_master_info)
+	    {
+	      er_init (er_msg_file, ER_NEVER_EXIT);
```

Comment:
why `er_init` here again?

Reply:
It is used for `er_set_with_oserror()`. 

when -a option is executed successfully, `db_shutdown ()` is executed. 
So we have to write `er_init()` to use this `er_set_with_oserror()`.

Reply:
er_init() has been already called in the line `3272`. What does `db_shutdown()` have to do with it?